### ISSUE TITLE
Failed to execute "npm install" command in Ubuntu server

### ISSUE THREAD

**Issue by jaballogian** [enketo/enketo-core#767]
**Describe the bug**
Failed to run "npm install" command in Ubuntu server

**To Reproduce**
Steps to reproduce the behavior:.
1. Testing in local server (Windows 10):
* Clone this repository using "git clone" command
* Go to "enketo-core" directory
* Run "npm install"
* Run "npm start" and everything was fine as seen in this screenshot below
![image](https://user-images.githubusercontent.com/24468466/110603592-c0e38080-81b9-11eb-9aec-382e7cb6df24.png)

2. Testing in Ubuntu server (Ubuntu 18.04.5 LTS)
* Clone this repository using "git clone" command
* Go to "enketo-core" directory
* Run "npm install" but failed as seen in the logs below

<details><summary>Error Logs</summary>
Here are the logs from the Ubuntu terminal:

```bash<!-- Syntax-highlighting:  paste your code below -->
user@ip:~/enketo-core$ ls
app.js  build  CHANGELOG.md  config.js  docs  Gruntfile.js  jsdoc.config.js  LICENSE  package.json  package-lock.json  README.md  rollup.config.js  src  test  tutorials

user@ip:~/enketo-core$ npm install

npm ERR! code 1
npm ERR! path /home/user/enketo-core/node_modules/node1-libxmljsmt
npm ERR! command failed
npm ERR! command sh -c node-pre-gyp install --fallback-to-build --loglevel http
npm ERR! make: Entering directory '/home/user/enketo-core/node_modules/node1-libxmljsmt/build'
npm ERR!   CXX(target) Release/obj.target/xmljs/src/libxmljs.o
npm ERR!   CXX(target) Release/obj.target/xmljs/src/xml_attribute.o
npm ERR!   CXX(target) Release/obj.target/xmljs/src/xml_document.o
npm ERR! xmljs.target.mk:180: recipe for target 'Release/obj.target/xmljs/src/xml_document.o' failed
npm ERR! make: Leaving directory '/home/user/enketo-core/node_modules/node1-libxmljsmt/build'
npm ERR! Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --loglevel=http --module=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release/xmljs.node --module_name=xmljs --module_path=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release --napi_version=7 --node_abi_napi=napi' (1)
npm ERR! node-pre-gyp http GET https://github.com/gagern/libxmljs/releases/download/v1.0.0-mt/node-v88-linux-x64.tar.gz
npm ERR! node-pre-gyp http 404 https://github.com/gagern/libxmljs/releases/download/v1.0.0-mt/node-v88-linux-x64.tar.gz
npm ERR! node-pre-gyp ERR! Tried to download(404): https://github.com/gagern/libxmljs/releases/download/v1.0.0-mt/node-v88-linux-x64.tar.gz
npm ERR! node-pre-gyp ERR! Pre-built binaries not found for node1-libxmljsmt@1.0.0 and node@15.8.0 (node-v88 ABI, glibc) (falling back to source compile with node-gyp)
npm ERR! node-pre-gyp http 404 status code downloading tarball https://github.com/gagern/libxmljs/releases/download/v1.0.0-mt/node-v88-linux-x64.tar.gz
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@7.1.2
npm ERR! gyp info using node@15.8.0 | linux | x64
npm ERR! gyp info ok
npm ERR! ../src/xml_document.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE libxmljs::XmlDocument::ToString(Nan::NAN_METHOD_ARGS_TYPE)’:
npm ERR! ../src/xml_document.cc:190:74: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          if (obj->Get(Nan::New<v8::String>("declaration").ToLocalChecked())->IsFalse()) {
npm ERR!                                                                           ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc:195:69: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          if (obj->Get(Nan::New<v8::String>("format").ToLocalChecked())->IsTrue()) {
npm ERR!                                                                      ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc:200:77: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          if (obj->Get(Nan::New<v8::String>("selfCloseEmpty").ToLocalChecked())->IsFalse()) {
npm ERR!                                                                              ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc:205:73: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          if (obj->Get(Nan::New<v8::String>("whitespace").ToLocalChecked())->IsTrue()) {
npm ERR!                                                                          ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE libxmljs::XmlDocument::FromXml(Nan::NAN_METHOD_ARGS_TYPE)’:
npm ERR! ../src/xml_document.cc:412:57: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          Nan::New<v8::String>("baseUrl").ToLocalChecked());
npm ERR!                                                          ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc:414:58: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          Nan::New<v8::String>("encoding").ToLocalChecked());
npm ERR!                                                           ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc: In member function ‘virtual void libxmljs::FromXmlWorker::WorkComplete()’:
npm ERR! ../src/xml_document.cc:521:31: warning: ‘v8::Local<v8::Value> Nan::Callback::Call(int, v8::Local<v8::Value>*) const’ is deprecated [-Wdeprecated-declarations]
npm ERR!          callback->Call(1, argv);
npm ERR!                                ^
npm ERR! In file included from ../src/libxmljs.h:7:0,
npm ERR!                  from ../src/xml_document.h:7,
npm ERR!                  from ../src/xml_document.cc:15:
npm ERR! ../../nan/nan.h:1742:3: note: declared here
npm ERR!    Call(int argc, v8::Local<v8::Value> argv[]) const {
npm ERR!    ^~~~
npm ERR! ../src/xml_document.cc:531:35: warning: ‘v8::Local<v8::Value> Nan::Callback::Call(int, v8::Local<v8::Value>*) const’ is deprecated [-Wdeprecated-declarations]
npm ERR!              callback->Call(1, argv);
npm ERR!                                    ^
npm ERR! In file included from ../src/libxmljs.h:7:0,
npm ERR!                  from ../src/xml_document.h:7,
npm ERR!                  from ../src/xml_document.cc:15:
npm ERR! ../../nan/nan.h:1742:3: note: declared here
npm ERR!    Call(int argc, v8::Local<v8::Value> argv[]) const {
npm ERR!    ^~~~
npm ERR! ../src/xml_document.cc:542:31: warning: ‘v8::Local<v8::Value> Nan::Callback::Call(int, v8::Local<v8::Value>*) const’ is deprecated [-Wdeprecated-declarations]
npm ERR!          callback->Call(2, argv);
npm ERR!                                ^
npm ERR! In file included from ../src/libxmljs.h:7:0,
npm ERR!                  from ../src/xml_document.h:7,
npm ERR!                  from ../src/xml_document.cc:15:
npm ERR! ../../nan/nan.h:1742:3: note: declared here
npm ERR!    Call(int argc, v8::Local<v8::Value> argv[]) const {
npm ERR!    ^~~~
npm ERR! ../src/xml_document.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE libxmljs::XmlDocument::userate(Nan::NAN_METHOD_ARGS_TYPE)’:
npm ERR! ../src/xml_document.cc:577:99: error: no matching function for call to ‘v8::Object::Set(v8::Local<v8::String>, v8::Local<v8::Array>)’
npm ERR!      info.Holder()->Set(Nan::New<v8::String>("userationErrors").ToLocalChecked(), errors.ToArray());
npm ERR!                                                                                                    ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3716:37: note: candidate: v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, v8::Local<v8::Value>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
npm ERR!                                      ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3716:37: note:   candidate expects 3 arguments, 2 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3719:37: note: candidate: v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, uint32_t, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
npm ERR!                                      ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3719:37: note:   candidate expects 3 arguments, 2 provided
npm ERR! ../src/xml_document.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE libxmljs::XmlDocument::Rnguserate(Nan::NAN_METHOD_ARGS_TYPE)’:
npm ERR! ../src/xml_document.cc:612:99: error: no matching function for call to ‘v8::Object::Set(v8::Local<v8::String>, v8::Local<v8::Array>)’
npm ERR!      info.Holder()->Set(Nan::New<v8::String>("userationErrors").ToLocalChecked(), errors.ToArray());
npm ERR!                                                                                                    ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3716:37: note: candidate: v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, v8::Local<v8::Value>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
npm ERR!                                      ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3716:37: note:   candidate expects 3 arguments, 2 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3719:37: note: candidate: v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, uint32_t, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
npm ERR!                                      ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3719:37: note:   candidate expects 3 arguments, 2 provided
npm ERR! make: *** [Release/obj.target/xmljs/src/xml_document.o] Error 1
npm ERR! gyp ERR! build error
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:378:20)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
npm ERR! gyp ERR! System Linux 4.15.0-124-generic
npm ERR! gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--loglevel=http" "--module=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release/xmljs.node" "--module_name=xmljs" "--module_path=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release" "--napi_version=7" "--node_abi_napi=napi"
npm ERR! gyp ERR! cwd /home/user/enketo-core/node_modules/node1-libxmljsmt
npm ERR! gyp ERR! node -v v15.8.0
npm ERR! gyp ERR! node-gyp -v v7.1.2
npm ERR! gyp ERR! not ok
npm ERR! node-pre-gyp ERR! build error
npm ERR! node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --loglevel=http --module=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release/xmljs.node --module_name=xmljs --module_path=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release --napi_version=7 --node_abi_napi=napi' (1)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/user/enketo-core/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:378:20)
npm ERR! node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1067:16)
npm ERR! node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
npm ERR! node-pre-gyp ERR! System Linux 4.15.0-124-generic
npm ERR! node-pre-gyp ERR! command "/usr/local/bin/node" "/home/user/enketo-core/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--loglevel" "http"
npm ERR! node-pre-gyp ERR! cwd /home/user/enketo-core/node_modules/node1-libxmljsmt
npm ERR! node-pre-gyp ERR! node -v v15.8.0
npm ERR! node-pre-gyp ERR! node-pre-gyp -v v0.9.1
npm ERR! node-pre-gyp ERR! not ok

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2021-03-10T07_49_48_830Z-debug.log

user@ip:~/enketo-core$ ls
app.js  build  CHANGELOG.md  config.js  docs  Gruntfile.js  jsdoc.config.js  LICENSE  package.json  package-lock.json  README.md  rollup.config.js  src  test  tutorials
```
</details>

**Expected behavior**
"npm install" and "npm run" are successful like the test in local (Windows) computer

**Local Computer Environment**
* OS: Windows 10 build 18363
* Node version: 12.19.0
* Npm version: 6.14.8
* Browser: Chrome version 88.0.4324.182

**Server Computer Environment**
* OS: Ubuntu 18.04.5 LTS (GNU/Linux 4.15.0-135-generic x86_64)
* Node version: 15.8.0
* Npm version: 7.5.4

**Comment by jaballogian** [comment]
Thank you for your response.

I installed build-essential in Ubuntu server using these steps:
* sudo apt-get update
* sudo apt-get install build-essential
* cd enketo-core/
* ls
* npm install

And here are the error logs (I thought the logs were still the same as before):

<details><summary>Error Logs</summary>
Here are the logs from the Ubuntu terminal:

```bash<!-- Syntax-highlighting:  paste your code below -->
user@ip:~$ sudo apt-get update
Hit:1 https://download.docker.com/linux/ubuntu bionic InRelease
Hit:2 http://archive.ubuntu.com/ubuntu bionic InRelease
Get:3 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
Get:4 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
Get:5 http://archive.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
Fetched 252 kB in 2s (155 kB/s)
Reading package lists... Done

user@ip:~$ sudo apt-get install build-essential
Reading package lists... Done
Building dependency tree
Reading state information... Done
build-essential is already the newest version (12.4ubuntu1).
0 upgraded, 0 newly installed, 0 to remove and 47 not upgraded.

user@ip:~$ cd enketo-core/

user@ip:~/enketo-core$ ls
app.js  build  CHANGELOG.md  config.js  docs  Gruntfile.js  jsdoc.config.js  LICENSE  package.json  package-lock.json  README.md  rollup.config.js  src  test  tutorials

user@ip:~/enketo-core$ npm install
npm ERR! code 1
npm ERR! path /home/user/enketo-core/node_modules/node1-libxmljsmt
npm ERR! command failed
npm ERR! command sh -c node-pre-gyp install --fallback-to-build --loglevel http
npm ERR! make: Entering directory '/home/user/enketo-core/node_modules/node1-libxmljsmt/build'
npm ERR!   CXX(target) Release/obj.target/xmljs/src/libxmljs.o
npm ERR!   CXX(target) Release/obj.target/xmljs/src/xml_attribute.o
npm ERR!   CXX(target) Release/obj.target/xmljs/src/xml_document.o
npm ERR! xmljs.target.mk:177: recipe for target 'Release/obj.target/xmljs/src/xml_document.o' failed
npm ERR! make: Leaving directory '/home/user/enketo-core/node_modules/node1-libxmljsmt/build'
npm ERR! Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --loglevel=http --module=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release/xmljs.node --module_name=xmljs --module_path=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release --napi_version=7 --node_abi_napi=napi' (1)
npm ERR! node-pre-gyp http GET https://github.com/gagern/libxmljs/releases/download/v1.0.0-mt/node-v88-linux-x64.tar.gz
npm ERR! node-pre-gyp http 404 https://github.com/gagern/libxmljs/releases/download/v1.0.0-mt/node-v88-linux-x64.tar.gz
npm ERR! node-pre-gyp ERR! Tried to download(404): https://github.com/gagern/libxmljs/releases/download/v1.0.0-mt/node-v88-linux-x64.tar.gz
npm ERR! node-pre-gyp ERR! Pre-built binaries not found for node1-libxmljsmt@1.0.0 and node@15.8.0 (node-v88 ABI, glibc) (falling back to source compile with node-gyp)
npm ERR! node-pre-gyp http 404 status code downloading tarball https://github.com/gagern/libxmljs/releases/download/v1.0.0-mt/node-v88-linux-x64.tar.gz
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@7.1.2
npm ERR! gyp info using node@15.8.0 | linux | x64
npm ERR! gyp info ok
npm ERR! ../src/xml_document.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE libxmljs::XmlDocument::ToString(Nan::NAN_METHOD_ARGS_TYPE)’:
npm ERR! ../src/xml_document.cc:190:74: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          if (obj->Get(Nan::New<v8::String>("declaration").ToLocalChecked())->IsFalse()) {
npm ERR!                                                                           ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc:195:69: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          if (obj->Get(Nan::New<v8::String>("format").ToLocalChecked())->IsTrue()) {
npm ERR!                                                                      ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc:200:77: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          if (obj->Get(Nan::New<v8::String>("selfCloseEmpty").ToLocalChecked())->IsFalse()) {
npm ERR!                                                                              ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc:205:73: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          if (obj->Get(Nan::New<v8::String>("whitespace").ToLocalChecked())->IsTrue()) {
npm ERR!                                                                          ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE libxmljs::XmlDocument::FromXml(Nan::NAN_METHOD_ARGS_TYPE)’:
npm ERR! ../src/xml_document.cc:412:57: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          Nan::New<v8::String>("baseUrl").ToLocalChecked());
npm ERR!                                                          ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc:414:58: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
npm ERR!          Nan::New<v8::String>("encoding").ToLocalChecked());
npm ERR!                                                           ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3763:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
npm ERR!    V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
npm ERR!                                            ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3766:43: note:   candidate expects 2 arguments, 1 provided
npm ERR! ../src/xml_document.cc: In member function ‘virtual void libxmljs::FromXmlWorker::WorkComplete()’:
npm ERR! ../src/xml_document.cc:521:31: warning: ‘v8::Local<v8::Value> Nan::Callback::Call(int, v8::Local<v8::Value>*) const’ is deprecated [-Wdeprecated-declarations]
npm ERR!          callback->Call(1, argv);
npm ERR!                                ^
npm ERR! In file included from ../src/libxmljs.h:7:0,
npm ERR!                  from ../src/xml_document.h:7,
npm ERR!                  from ../src/xml_document.cc:15:
npm ERR! ../../nan/nan.h:1742:3: note: declared here
npm ERR!    Call(int argc, v8::Local<v8::Value> argv[]) const {
npm ERR!    ^~~~
npm ERR! ../src/xml_document.cc:531:35: warning: ‘v8::Local<v8::Value> Nan::Callback::Call(int, v8::Local<v8::Value>*) const’ is deprecated [-Wdeprecated-declarations]
npm ERR!              callback->Call(1, argv);
npm ERR!                                    ^
npm ERR! In file included from ../src/libxmljs.h:7:0,
npm ERR!                  from ../src/xml_document.h:7,
npm ERR!                  from ../src/xml_document.cc:15:
npm ERR! ../../nan/nan.h:1742:3: note: declared here
npm ERR!    Call(int argc, v8::Local<v8::Value> argv[]) const {
npm ERR!    ^~~~
npm ERR! ../src/xml_document.cc:542:31: warning: ‘v8::Local<v8::Value> Nan::Callback::Call(int, v8::Local<v8::Value>*) const’ is deprecated [-Wdeprecated-declarations]
npm ERR!          callback->Call(2, argv);
npm ERR!                                ^
npm ERR! In file included from ../src/libxmljs.h:7:0,
npm ERR!                  from ../src/xml_document.h:7,
npm ERR!                  from ../src/xml_document.cc:15:
npm ERR! ../../nan/nan.h:1742:3: note: declared here
npm ERR!    Call(int argc, v8::Local<v8::Value> argv[]) const {
npm ERR!    ^~~~
npm ERR! ../src/xml_document.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE libxmljs::XmlDocument::userate(Nan::NAN_METHOD_ARGS_TYPE)’:
npm ERR! ../src/xml_document.cc:577:99: error: no matching function for call to ‘v8::Object::Set(v8::Local<v8::String>, v8::Local<v8::Array>)’
npm ERR!      info.Holder()->Set(Nan::New<v8::String>("userationErrors").ToLocalChecked(), errors.ToArray());
npm ERR!                                                                                                    ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3716:37: note: candidate: v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, v8::Local<v8::Value>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
npm ERR!                                      ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3716:37: note:   candidate expects 3 arguments, 2 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3719:37: note: candidate: v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, uint32_t, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
npm ERR!                                      ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3719:37: note:   candidate expects 3 arguments, 2 provided
npm ERR! ../src/xml_document.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE libxmljs::XmlDocument::Rnguserate(Nan::NAN_METHOD_ARGS_TYPE)’:
npm ERR! ../src/xml_document.cc:612:99: error: no matching function for call to ‘v8::Object::Set(v8::Local<v8::String>, v8::Local<v8::Array>)’
npm ERR!      info.Holder()->Set(Nan::New<v8::String>("userationErrors").ToLocalChecked(), errors.ToArray());
npm ERR!                                                                                                    ^
npm ERR! In file included from /home/user/.cache/node-gyp/15.8.0/include/node/node.h:63:0,
npm ERR!                  from ../src/xml_document.cc:3:
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3716:37: note: candidate: v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, v8::Local<v8::Value>, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
npm ERR!                                      ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3716:37: note:   candidate expects 3 arguments, 2 provided
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3719:37: note: candidate: v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, uint32_t, v8::Local<v8::Value>)
npm ERR!    V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
npm ERR!                                      ^~~
npm ERR! /home/user/.cache/node-gyp/15.8.0/include/node/v8.h:3719:37: note:   candidate expects 3 arguments, 2 provided
npm ERR! make: *** [Release/obj.target/xmljs/src/xml_document.o] Error 1
npm ERR! gyp ERR! build error
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:378:20)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
npm ERR! gyp ERR! System Linux 4.15.0-124-generic
npm ERR! gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--loglevel=http" "--module=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release/xmljs.node" "--module_name=xmljs" "--module_path=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release" "--napi_version=7" "--node_abi_napi=napi"
npm ERR! gyp ERR! cwd /home/user/enketo-core/node_modules/node1-libxmljsmt
npm ERR! gyp ERR! node -v v15.8.0
npm ERR! gyp ERR! node-gyp -v v7.1.2
npm ERR! gyp ERR! not ok
npm ERR! node-pre-gyp ERR! build error
npm ERR! node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --loglevel=http --module=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release/xmljs.node --module_name=xmljs --module_path=/home/user/enketo-core/node_modules/node1-libxmljsmt/build/Release --napi_version=7 --node_abi_napi=napi' (1)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/user/enketo-core/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:378:20)
npm ERR! node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1067:16)
npm ERR! node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
npm ERR! node-pre-gyp ERR! System Linux 4.15.0-124-generic
npm ERR! node-pre-gyp ERR! command "/usr/local/bin/node" "/home/user/enketo-core/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--loglevel" "http"
npm ERR! node-pre-gyp ERR! cwd /home/user/enketo-core/node_modules/node1-libxmljsmt
npm ERR! node-pre-gyp ERR! node -v v15.8.0
npm ERR! node-pre-gyp ERR! node-pre-gyp -v v0.9.1
npm ERR! node-pre-gyp ERR! not ok

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2021-03-12T02_12_27_557Z-debug.log

user@ip:~/enketo-core$ ls
app.js  build  CHANGELOG.md  config.js  docs  Gruntfile.js  jsdoc.config.js  LICENSE  package.json  package-lock.json  README.md  rollup.config.js  src  test  tutorials

```
</details>

**Comment by MartijnR** [comment]
I think I know what's going on. Both the npm version and node version are likely not supported by libxslt, so they should be 6 and 12 respectively. This library is always a problem, and it may take years (or weeks or days) before we can upgrade node and npm. Out of our control unfortunately.

**Comment by jaballogian** [comment]
Thank you for your response.
I will try to use npm and node versions to 6 and 12 respectively.

**Comment by lognaturel** [comment]
Closing as duplicate of https://github.com/enketo/enketo-transformer/issues/98 unless we hear otherwise!
